### PR TITLE
Add example on how to use NLocal 

### PR DESCRIPTION
**Pull request by ghost**
### Summary
this change is to add an example showing how to quickly prototype variational circuits using NLocal class.

### Details and comments
This example shows the most common workflow cases for Variational or parametrized quantum circuits.

### CODE REVIEW

**Comment by ShellyGarion** [comment]
@shef4 - thanks for your contribution to qiskit, but this PR needs some work in order to be merged to qiskit.
First, could you please sign the CLA?
I would suggest that you look at the [contributing guidelines](https://github.com/Qiskit/qiskit/blob/main/CONTRIBUTING.md) and see how to make a PR that fits to the style and documentation format of qiskit.
In particular, if you would like to add examples for the `NLocal` class, then they should be added in the class docstring, see e.g. the examples in the [`TwoLocal`](https://github.com/Qiskit/qiskit/blob/stable/1.1/qiskit/circuit/library/n_local/two_local.py#L76-L136) class.

**Comment by 1ucian0** [comment]
@shef4 is a first time contributor to the repo. Let's give them some extra time to answer the @ShellyGarion comments before closing this as stalling.

**Comment by ghost** [comment]
@ShellyGarion @1ucian0 Thank you for the help and patience. I signed the CLA, and will look at the example you shared of the [TwoLocal](https://github.com/Qiskit/qiskit/blob/stable/1.1/qiskit/circuit/library/n_local/two_local.py#L76-L136) class to update the docstring. I will also update this PR or make a new one that follows the style and documentation.

**Comment by 1ucian0** [comment]
ping @shef4 ? Still working on it? (no worries if the answer is yes, just checking if it should be closed)

**Comment by ghost** [comment]
Yes, thanks for the reminder! its a bit lengthy but  I wanted to make it readable for newer users. Happy to make changes as need or to make it match [TwoLocal](https://github.com/Qiskit/qiskit/blob/stable/1.1/qiskit/circuit/library/n_local/two_local.py#L76-L136) more

File: qiskit/circuit/library/n_local/n_local.py [hunk 1]
```diff
@@ -66,6 +66,94 @@ class NLocal(BlueprintCircuit):
 
     If specified, barriers can be inserted in between every block.
     If an initial state object is provided, it is added in front of the NLocal.
+
+    Examples:
+        >>> q0,q1,q2,q3 = 0,1,2,3
+
+        >>> num_qubits = 3
+        >>> rotation_blocks = 'ry'      # this could be a circuit, list of circuit, circuit instruction ('ry'), list of circuit instructions, or None for default values
+        >>> entanglement_blocks = 'cx'  # this could be a circuit, list of circuit, circuit instruction ('ry'), list of circuit instructions, or None for default values
```
Comment (ShellyGarion) [inline qiskit/circuit/library/n_local/n_local.py]: I suspect these two lines are too long. In addition, the comment in the second one should be with cx.

**Review by ShellyGarion — COMMENTED** [review]
Hi [shef4](https://github.com/shef4) - your contribution looks much better now, but still needs some more work.

1) Could you please fix the lint errors? (see the [style and lint section](https://github.com/Qiskit/qiskit/blob/main/CONTRIBUTING.md#style-and-lint) in the contributor guidelines.

2) I see that all the examples you added also appear in the [`TwoLocal` class API docs](https://docs.quantum.ibm.com/api/qiskit/0.24/qiskit.circuit.library.TwoLocal).
Are there some examples that one should specifically use the `NLocal` class and not the `TwoLocal` class?

**Comment by ghost** [comment]
Hi @ShellyGarion, 

I checked the documentation and saw that [Nlocal](https://docs.quantum.ibm.com/api/qiskit/0.24/qiskit.circuit.library.NLocal) has 1 extra parameter than [TwoLocal](https://docs.quantum.ibm.com/api/qiskit/0.24/qiskit.circuit.library.TwoLocal): `overwrite_block_parameters=True`

I'll make the changes to fix the lint test but look more into how `overwrite_block_parameters` could be used. 

Thanks!

**Comment by ShellyGarion** [comment]
thanks @shef4 - I think that the main difference between the `TwoLocal` and the `NLocal` classes is that `TwoLocal` is aimed to handle 2-qubit entangling gates, while the `NLocal` class can handle any n-qubit entangling gate. So it might be interesting to show examples with n-qubit gates where n>2, such as the Toffoli `CCX` gate.

**Comment by ghost** [comment]
@ShellyGarion, I'll update the examples to show support for multi-qubit circuits and gates like `CCX` and will close the PR for now as I want to find time to properly go through the lint doc test. 

Thanks a lot for the help through this process! 
1 step closer to my 1st PR in this repo :)

**Pull request closed without merging**